src/bootstrap/src/core/download.rs

```diff
@@ -250,19 +257,113 @@ impl Config {
                             "(New-Object System.Net.WebClient).DownloadFile('{}', '{}')",
                             url, tempfile.to_str().expect("invalid UTF-8 not supported with powershell downloads"),
                         ),
-                    ])).is_err() {
+                    ])).is_ok() {
                         return;
                     }
                     eprintln!("\nspurious failure, trying again");
                 }
             }
             if !help_on_error.is_empty() {
                 eprintln!("{help_on_error}");
+                Self::check_outdated(commit);
             }
             crate::exit!(1);
         }
     }
 
+    fn check_outdated(commit: &String) {
+        let check_outdated_msg = || {
+            if !commit.is_empty() {
```

Comment:
It seems surprising to do all of the user.name-based work, and then end up not actually looking at any timestamps anyway because `commit.is_empty()`.

Reply:
`commit.is_empty()` is really just a sanity check, it should be very unlikely to hit that branch of the if statement now. Not sure how this should be refactored.